docs/docs/tutorials/noirjs_app.md

```diff
@@ -248,119 +141,160 @@ At this point in the tutorial, your folder structure should look like this:
-
 :::
 
-## Some NoirJS
+## Compile compile compile
 
-We're starting with the good stuff now. If you've compiled the circuit as described above, you should have a `json` file we want to import at the very top of our `main.js` file:
+Finally we're up for something cool. But before we can execute a Noir program, we need to compile it into ACIR: an abstract representation. Here's where `noir_wasm` comes in.
 
-```ts
-import circuit from '../circuit/target/circuit.json';
+`noir_wasm` expects a filesystem so it can resolve dependencies. While we could use the `public` folder, let's just import those using the nice `?url` syntax provided by vite. At the top of the file:
+
+```js
+import { compile, createFileManager } from "@noir-lang/noir_wasm"
+
+import main from "./circuit/src/main.nr?url";
+import nargoToml from "./circuit/Nargo.toml?url";
 ```
 
-[Noir is backend-agnostic](../index.mdx#whats-new-about-noir). We write Noir, but we also need a proving backend. That's why we need to import and instantiate the two dependencies we installed above: `BarretenbergBackend` and `Noir`. Let's import them right below:
+Compiling on the browser is common enough that `createFileManager` already gives us a nice in-memory filesystem we can use. So all we need to compile is fetching these files, writing them to our filesystem, and compile. Add this function:
 
 ```js
-import { BarretenbergBackend, BarretenbergVerifier as Verifier } from '@noir-lang/backend_barretenberg';
+export async function getCircuit() {
+ const fm = createFileManager("/");
+ const { body } = await fetch(main);
+ const { body: nargoTomlBody } = await fetch(nargoToml);
+
+ fm.writeFile("./src/main.nr", body);
+ fm.writeFile("./Nargo.toml", nargoTomlBody);
+ return await compile(fm);
+}
+```
+
+:::tip
+
+As you can imagine, with `node` it's all conveniently easier since you get native access to `fs`...
+
+:::
+
+## Some more JS
+
+We're starting with the good stuff now. We want to execute our circuit to get the witness, and then feed that witness to Barretenberg. Luckily, both packages are quite easy to work with. Let's import them at the top of the file:
+
+```js
+import { UltraHonkBackend } from '@aztec/bb.js';
 import { Noir } from '@noir-lang/noir_js';
 ```
 
 And instantiate them inside our try-catch block:
 
 ```ts
 // try {
-const backend = new BarretenbergBackend(circuit);
-const noir = new Noir(circuit);
+const { program } = await getCircuit();
+const noir = new Noir(program);
+const backend = new UltraHonkBackend(program.bytecode);
 // }
 ```
 
-:::note
+:::warning
 
-For the remainder of the tutorial, everything will be happening inside the `try` block
+WASMs are not always easy to work with. In our case, `vite` likes serving them with the wrong MIME type. There are different fixes but we found the easiest one is just YOLO instantiating the WASMs manually. Paste this at the top of the file, and it will work just fine:
```

Comment:
```suggestion
WASMs are not always easy to work with. In our case, `vite` likes serving them with the wrong MIME type. There are different fixes but we found the easiest one is just YOLO instantiating the WASMs manually. Paste this at the top of the file, just below the other imports, and it will work just fine:
```